src/hooks/usePreviousModified.js

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useRef } from "react";
+
+const usePreviousModified = (value, predicate) => {
+	const ref = useRef();
+
+	useEffect(() => {
+		ref.current = value;
+	}, [value]);
+
+	return predicate
+		? predicate(ref.current, value)
+		: ref.current !== undefined && ref.current !== null && ref.current !== value;
```

Comment:
Better done as a default value for `predicate`, see `useLoader` and its `cutoutSelector` parameter for inspiration.